dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPUtils.java

```diff
@@ -0,0 +1,254 @@
+        ServerBootstrapExtended bs = new ServerBootstrapExtended();
+        bs.group(framework.getParentGroup(), framework.getChildGroup());
+        bs.channel(NioServerSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    /**
+     * Create a {@link BootstrapExtended} for outbound TCP channels
+     * 
+     * @param framework
+     * @param tcpOptions
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createTCPBootstrapOutbound(NettyFrameworkImpl framework,
+            Map<String, Object> tcpOptions) throws NettyException {
+        BootstrapConfiguration config = new TCPConfigurationImpl(tcpOptions, false);
+        BootstrapExtended bs = new BootstrapExtended();
+        bs.group(framework.getParentGroup());
+        bs.channel(NioSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    private static ChannelFuture open(NettyFrameworkImpl framework, AbstractBootstrap bootstrap,
+            final TCPConfigurationImpl config, String inetHost, int inetPort, ChannelFutureListener openListener,
+            final int retryCount) {
+
+        ChannelFuture oFuture = null;
+        if (inetHost.equals("*")) {
+            inetHost = "0.0.0.0";
+        }
+        if (config.isInbound()) {
+            oFuture = ((ServerBootstrapExtended) bootstrap).bind(inetHost, inetPort);
+        } else {
+            oFuture = ((BootstrapExtended) bootstrap).connect(inetHost, inetPort);
+        }
+        final ChannelFuture openFuture = oFuture;
+        if (openListener != null) {
+            openFuture.addListener(openListener);
+        }
+        final String newHost = inetHost;
+
+        openFuture.addListener(future -> {
+            if (future.isSuccess()) {
+
+                // add the new channel to the set of active channels, and set a close future to
+                // remove it
+                final Channel channel = openFuture.channel();
+                framework.getActiveChannels().add(channel);
+                channel.closeFuture().addListener(innerFuture -> framework.stop(channel));
+
+                // set common channel attrs
+                channel.attr(ConfigConstants.NameKey).set(config.getExternalName());
+                channel.attr(ConfigConstants.HostKey).set(newHost);
+                channel.attr(ConfigConstants.PortKey).set(inetPort);
+                channel.attr(ConfigConstants.IsInboundKey).set(config.isInbound());
+
+                // set up the a helpful log message
```

Comment:
Remove 'the' ? Also in the following lines change 0.0.0.0, IPv4, IPv6 to constants

Reply:
changed, except ipxx constants...they are used only once here and in the equivalent udp fn. Not sure that's worth it.